### ISSUE TITLE
Benchmarking model speed

### ISSUE THREAD

**Issue by jscanvic** [deepinv/deepinv#1212]
### Describe the feature

I suggest we add ways to benchmark model speed in the library.

Below are some design options.

**Metrics**
- FPS (Hz): average number of batch entries processed per second
- Inference speed (s): average time to process a batch entry
- Throughput (bit/s): average number of bits processed per second where N_bits = N_pixels * bits per pixel
- FLOPs: number of floating point operations - thanks Romain

**Modes**
- Inference (forward-pass only)
- Training (forward-pass and backward-pass)
- Backward (backward-pass only)

**Interface**
- A standalone benchmarking function / class that can be called directly on a model and a dataset, e.g.,
```python
import deepinv as dinv

model = ...
dataset = ...
result = dinv.utils.benchmark(model, dataset)
```
- Integration with the trainer as an additional eval / test metric

**Implementation**
- Warm-up iterations
- Proper cuda synchronization
- Include some form of hardware info (GPU / CPU models)

**Related**
- It could be a neat addition to the future leaderboard #1002 

### Motivation

/

**Comment by romainvo** [comment]
Complementary, but feel like it belongs to the same kind of info, would be to provide the FLOPs of a model/algorithm and a method to compute it. I know there's this (https://meta-pytorch.org/tnt/stable/utils/generated/torchtnt.utils.flops.FlopTensorDispatchMode.html)  but I haven't digged into it

**Comment by Tmodrzyk** [comment]
I feel like for a fair comparison these performance benchmarks should run on approximately similar nodes.
A simple solution would be to use the node currently dedicated to the GPU CI, but I fear this would overload it and slow down CI further. 
Maybe we can find a way to somehow combine it with the GPU tests ?

**Comment by tachella** [comment]
good idea! The [benchmarks](https://github.com/deepinv/deepinv/pull/1002) should provide some timing information as well
